core/src/main/scala/org/broadinstitute/dsde/rawls/entities/EntityService.scala

```diff
@@ -413,13 +413,21 @@ class EntityService(protected val ctx: RawlsRequestContext,
           val newAccum = prev.accum ++ curr.accum
           AttrAccum(newAccum, None)
 
-        // midstream, we notice that the current entity is DIFFERENT from the previous entity.
+        // midstream, we notice that the current entity's id is greater than the previous entity's id.
         // take all the attributes we have gathered for the previous entity,
         // marshal them into an Entity object, emit that Entity, and start a new accumulator
         // for the new/current entity
-        case (prev: AttrAccum, curr: AttrAccum) if prev.accum.head.entityRecord.id != curr.accum.head.entityRecord.id =>
+        case (prev: AttrAccum, curr: AttrAccum) if prev.accum.head.entityRecord.id < curr.accum.head.entityRecord.id =>
           entityFinished(prev.accum, curr.accum)
 
+        // midstream, we notice that the current entity's id is LESS than the previous entity's id.
+        // this breaks the assumption that entities are ordered by their ids ascending, and indicates a coding
+        // error. Throw an exception.
+        case (prev: AttrAccum, curr: AttrAccum) if prev.accum.head.entityRecord.id > curr.accum.head.entityRecord.id =>
```

Comment:
ah, this was discussed at length in the first PR, see https://github.com/broadinstitute/rawls/pull/2247#discussion_r1120767592. It will only happen if a future Rawls developer changes the database query, which is definitely a hypothetical. Do you think I should leave this clause out?

Reply:
I think it's good. I think my vote would be to catch it through unit tests if this did happen, but this is an improvement, and I don't think it's worth the effort of changing it to unit testing when we have this logic written in this PR :)